src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -452,13 +461,11 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
     __ lea(rsi, Address(rbp, in_bytes(VM_Version::std_cpuid1_offset())));
     __ movl(rcx, 0x18000000); // cpuid1 bits osxsave | avx
     __ andl(rcx, Address(rsi, 8)); // cpuid1 bits osxsave | avx
-    __ cmpl(rcx, 0x18000000);
-    __ jccb(Assembler::notEqual, done); // jump if AVX is not supported
+    __ jccb(Assembler::equal, done); // jump if AVX is not supported
```

Comment:
This doesn't not have same effect as before. Consider input is 0x10000000, the andl result will not be zero with this code and so jump to done will not happen. Whereas prior to this change, the cmpl with 0x18000000 will fail for equality and so a jump to done will happen.  This is the case for all the places where we are checking more than 1 set bit.

Reply:
Thanks @sviswa7 , sub-optimality was mainly around single-bit comparisons, where we could save redundant CMP after AND, and by flipping the predicate of subsequent flag-consuming JMP,  multibits should remain unaltered.